tests/test_ranges.py

```diff
@@ -617,3 +617,135 @@ def test_days_property_has_correct_value(self):
             end=datetime.date(2000, 1, 2),
         )
         assert range.days == 2
+
+    class TestUnion:
+        @pytest.mark.parametrize(
+            "range, other", [
+                (
+                    ranges.FiniteDateRange(
+                        start=datetime.date(2000, 1, 1),
+                        end=datetime.date(2000, 1, 2),
+                    ),
+                    ranges.FiniteDateRange(
+                        start=datetime.date(2000, 1, 3),
+                        end=datetime.date(2000, 1, 4),
+                    ),
+                ),
+                (
+                    ranges.FiniteDateRange(
+                        start=datetime.date(2000, 1, 3),
+                        end=datetime.date(2000, 1, 4),
+                    ),
+                    ranges.FiniteDateRange(
+                        start=datetime.date(2000, 1, 1),
+                        end=datetime.date(2000, 1, 2),
+                    ),
+                ),
+            ]
+        )
+        def test_handles_adjacent_ranges(self, range, other):
+            """
+            As FiniteDateRange boundaries are double inclusive, when two ranges
+            end/start on consecutive days they effectively cover a fully contiguous
+            period of time and should make a union.
+            """
+            union = range | other
+            assert union == ranges.FiniteDateRange(
+                start=datetime.date(2000, 1, 1),
+                end=datetime.date(2000, 1, 4),
+            )
+
+        def test_handles_ranges_out_of_order(self):
```

Comment:
:panda_face: I don't follow - isn't this identical to the 2nd case for the test above?

Reply:
Yeah I think I meant to delete this in a test refactor, my bad.